### PR TITLE
Adjust range meta key validation to allow for max key.

### PR DESCRIPTION
**Pull request by spencerkimball**

### CODE REVIEW

File: server/server_test.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ import (
 
 	"golang.org/x/net/context"
 
-	"github.com/cockroachdb/c-snappy"
+	snappy "github.com/cockroachdb/c-snappy"
```
Comment (tschottdorf) [inline server/server_test.go]: was this an automated change?

**Comment by tschottdorf** [comment]
LGTM

File: server/server_test.go (hunk 1 above)
Comment (spencerkimball) [inline server/server_test.go]: My goimports kept removing it.

**Comment by tschottdorf** [comment]
@spencerkimball: @tamird and I got confused about this. Could you explain this a little bit? 

> // Body of the key must sort <= KeyMax. KeyMax might be the body in
// the event of doing a lookup of the key "\x00\x00meta1\xff\xff" in
// order to resolve an intent during a split.

- isn't `\x00\x00meta1\xff\xff` always on the first range?
- if we're talking about `\x00\x00meta2\xff\xff`: 
if I'm trying to look up the range containing `\x00\x00meta2\xff\xff`, wouldn't your RangeMetaKey be
`\x00\x00meta1\x00\x00meta2\xff\xff` with **body** `\x00\x00meta2\xff\xff`, which is perfectly between `KeyMin` and `KeyMax`?

**Comment by spencerkimball** [comment]
Yes, you're correct--this change shouldn't be necessary. The dist sender should never send such a range lookup request.

I'm not sure what motivated this change, and it's annoying that I can't remember if there was a more specific reason than me just being wrong about some code inspection. I'll back it out.

**Comment by spencerkimball** [comment]
Oh wait, I take that back. It is the case of `\x00\x00meta2\xff\xff`.

The problem is that that is translated into `\x00\x00meta1\xff\xff`, not `\x00\x00meta1\x00\x00meta2\xff\xff` (refer to keys/keys.go:243)

**Comment by tschottdorf** [comment]
Ok, took a good look at it and everything makes sense (again?). For a bit I wondered whether this only affects the `\x00\x00meta2\xff\xff` descriptor (or others with something lower than `\xff\xff` as well) but that's not the case (the key ranges described by the meta records are exclusive at the start key, inclusive at the end key, and this commit corresponds to `\infinity+1 == \infinity`.

**Pull request merged**